torch/autograd/gradcheck.py

```diff
@@ -266,7 +266,11 @@ def fail_test(msg):
                     'This check will likely fail if all the inputs are '
                     'not of double precision floating point or complex. ')
             content = inp._values() if inp.is_sparse else inp
-            if content.layout is not torch._mkldnn and any([s == 0 for s in content.stride()]):
+            # TODO: To cover more problematic cases, replace stride = 0 check with
+            # "any overlap in memory" once we have a proper function to check it.
+            if content.layout is not torch._mkldnn and \
+               content.nelement() > 1 and \
+               any([s == 0 for s in content.stride()]):
```

Comment:
Tensor with sizes (2,1,3) and strides (3,0,1) is fine, but would fail your check. Strides where size=1 should be ignored altogether.

Reply:
how about just `not all(st > 0 or sz <= 1 for st, sz in zip(content.stride(), content.size()))`?

Reply:
Yes looks much better !